### ISSUE TITLE
Fallout 4 Support

### ISSUE THREAD

**Issue by DiglidiDudeNG** [TanninOne/modorganizer#352]
When trying to link MO with the installation folder of Fallout 4, it does not detect it as a compatible game.

I know the game just came out yesterday, however there is no way I'm going to clutter its data folder with mods.

**Comment by tjmidnight** [comment]
There are no mods to clutter it's folder with at the moment.
@Tannin: I'd like to throw my support behind a FO4 version of MO. Myself and the TTW team would be happy to test if needed.

**Comment by master412160** [comment]
I think it would be cool to do it.

**Comment by TanninOne** [comment]
Fallout 4 is a 64-bit game and as such mo is not compatible with it.

I've been working on 64-bit support for a while and in the last few months this was practically the only thing i've been working on.

My hope is that i will have a working alpha versions within a few weeks.

But please be aware that this is a complete rewrite of the whole virtualisation functionality and its more complex than before so the first releases will probably be less reliable.

Therefore I will be very thankful for volunteer testers but if you just want to play and need a stable mod manager you may need a bit of patience.

But since fo4 has a new engine and requires new tools I think fo4 modding will take a while to get going anyway.

**Comment by tjmidnight** [comment]
Myself and at least several members of my team are at your disposal when needed.

**Comment by pralima87** [comment]
This is my nexus profile:
http://www.nexusmods.com/skyrim/users/1867404/?
I can help in the testing ;)

**Comment by zax-ftw** [comment]
Since you are rewriting anyway... It would be awesome if you could make one installation support multiple games this time. Having to install same app twice (for Skyrim and Fallout 4) feels clunky.

**Comment by Freso** [comment]
I will be happy to test anything guaranteed to break my setup! :+1:

**Comment by WalrusWay** [comment]
I like MO as simple and clean as possible.  I use it with FO3, FNV, and  Skyrim.  I much prefer having a copy of the MO executable in each top dir that only sees that game and only its subdirs.  To me it is a utility I add to a game.  Running multiple mod profiles for a single game to support multiple chars (like a set of archery mods for an archer char or Khajiit mods for a Khajiit char) already gets chaotic enough.  One big MO installation to support multiple games is not something I want or crave--- perhaps when game companies release their games that way (one big installer for all their games) then I'd organize how I mod that way.  Beyond that, count me in  for x64 testing once I complete my vanilla playthrough!

**Comment by zax-ftw** [comment]
I know its humans nature to hate changes, even if they are for better, but I don't understand how an option to switch game profiles on fly would make it more chaotic for you?
 
Having everything in one place, using one shortcut - its anything, but chaotic, lol!

Game specific profiles would still exist, mod folders would be still separate, basically nothing would change with exception of shared UI and better mod folders organization on disk.

**Comment by ralexpdx** [comment]
@TanninOne, I for one would no longer play FONV or Skyrim without MO. I can't wait to get MO for FO4 so that things work as mod authors intend. I have been SO happy with the abilities of MO to allow me to play these games the way I want to and will sit on the sidelines cheering you and your team on for compatibility with FO4. Go team!

**Comment by nekoyoubi** [comment]
@TanninOne I would be more than willing to test as well. Thanks for keeping us all in the loop.

Nexus (if needed): http://www.nexusmods.com/games/users/5040332/?

**Comment by MithosFall** [comment]
I'll go ahead and throw my hat in the ring as well.  I would love to be a tester for MO for FO4.  My Nexus account: http://www.nexusmods.com/games/users/1658883/

**Comment by nar001** [comment]
Same here, I'd be happy to help ! Here my Nexus account ! www.nexusmods.com/games/users/3538508/?

**Comment by DAOWAce** [comment]
A full rewrite for handling 64bit games in regards to virtualization of folders?

..I don't understand.  That's an OS level thing, and MO is clearly working on 64bit Windows.

**Comment by nar001** [comment]
@DAOWAce I'm pretty sure he's talking about Fallout 4 being 64 bits, and thus needing a rewrite since the old MO is 32 bits only

**Comment by TanninOne** [comment]
@DAOWAce windows provides backwards compatibility so 32 bit applications can run on 64bit Windows without the author doing anything. That's different from the application taking advantage of 64bit addressing.
MOs filesystem virtualisation is very low-level stuff however where windows' backwards compatibility doesn't help, it has to work with the same bitness as the application.

**Comment by wmcook32** [comment]
Ever since discovering MO I basically refused to play any game with NMM. That being said I would love to help test the 64-bit version. My Nexus profile: http://www.nexusmods.com/fallout4/users/6142045/?

**Comment by LawLeewer** [comment]
I would be glad to help testing the 64-bit version!  My nexus profile: http://www.nexusmods.com/fallout4/users/13791360/?

**Comment by burntcheese** [comment]
I dislike NMM, and MO is an essential for me when it comes to modding TES and FO. I'd love to test it out. Nexus profile is http://www.nexusmods.com/fallout4/users/5034165/?

**Comment by cplr** [comment]
I'm also confused what FO4 being a 64-bit binary has to do with anything when we are talking managing the file system here? Can anyone explain with technical details?

**Comment by DAOWAce** [comment]
@TanninOne So it's deeper/different than just extracting archives and symlinking folders/files to the data directory?  I've been symlinking for years and can't understand why something that functions the same would require a full rewrite just for a 64bit application; the app doesn't actually matter unless it specifically can't use symlinked files/folders.

'Course, I don't do any sort of coding, so I don't know anything further in regards to MO's form of symlinks. Is there a writeup you have of the technical details?

**Comment by TanninOne** [comment]
MO isn't using symlinks provided by the ntfs for a couple of reasons. It's closer to a driver that provides its own link mechanisms.
Advantages:
* My links can "override" real files in the destination without deleting
* They work independent of the fs "i.e from a fat formatted usb stick"
* They don't require admin rights to create
* They are far quicker to create
* They are visible only to the processes you choose.
* You can map a writable directory onto a real dir, thus capturing all newly created files (for the overwrite mod)

**Comment by TanninOne** [comment]
Technical details: I don't have a writeup. In short: I hook into the low-level windows api functions that deal with file names (mostly those are resolving/enumerating file names) and manipulate their call parameters and return values in such a way that the mapped files appear to exist in the intended location.
This happens putely inside a processes memory so no permissions above the process permissions are required and there is no risk of "damaging" anything outside the process rights.

**Comment by DAOWAce** [comment]
Interesting, had no idea a lower level 'symlink' system even existed.

Thanks for educating me in what MO does and what's possible. :)

**Comment by koumdros** [comment]
I'm also available for testing. Experienced user of MO with all bethesda games, runing multiple setups for years.

**Comment by TheGreenDragon** [comment]
Considering how much I love this software, I would be more than happy to help testing. 
http://www.nexusmods.com/fallout4/users/1617465/?

**Comment by Newyears1978** [comment]
Looking forward to it, I don't like to mod without MO. MO is the best, the file virtualisation is awesome, so much cleaner and easier than Manual/NMM or any other method.

**Comment by NoShotz** [comment]
I have been installing mods manually since i absolutely hate how NMM looks, can't wait for MO to support fallout 4

**Comment by TanninOne** [comment]
Update: I've been playing fo4 with the new vfs lib and a handful of mods for a few hours now and to my surprise it appears to be fully stable.

There are 2 known bugs in the lib that dont appear to affect the game but may become a problem with third party tools and support for 32-bit (for older games) is untested.

My build process is currently not succesfully building 2 of the mo plugins and doesnt create an installer yet.

Local savegames arent supported yet, neither are ini tweaks.

And i have yet to assign a license to the new lib.

i think we can live wirh some of that for now and the rest i should be able to fix soon so I estimate a first alpha in the next 1-2 days.

**Comment by RERLTuna** [comment]
Would love to test the new ModOrganizer. Used it for all Skyrim, went back to FNV/FO3 and had to figure out the tricks to get it to work there because its just like a magic mod manager. Anything I can do to help let me know. http://www.nexusmods.com/fallout4/users/2892099/?

**Comment by NoShotz** [comment]
I would gladly test the alpha when it comes out, http://www.nexusmods.com/fallout4/users/8671081/

**Comment by Freso** [comment]
> And i have yet to assign a license to the new lib.

Please don't go for something more restrictive than LGPL for libraries. :)

Looking through some of the repositories, it looks like several of them do not have a defined license though?

**Comment by CatmanIX** [comment]
I'd also be down with testing if this isn't already fully supported by the time i actually get to playing FO4 (which won't be until after the official modding tool is released at the earliest).

**Comment by pralima87** [comment]
Great news Tannin!
A alpha will already be great news!

**Comment by strangerism** [comment]
I will be happy to test the beta.

http://www.nexusmods.com/fallout4/users/116794/?

**Comment by TanninOne** [comment]
I've just uploaded a first alpha version, please give feedback here. https://github.com/TanninOne/modorganizer/releases/tag/2.0alpha1

**Comment by nar001** [comment]
Woops, I'm really tired and posted in the "issues" thing of Github, sorry ! I have an error at startup, I took a screenshot ! 
![bug](https://cloud.githubusercontent.com/assets/15905072/11350679/00949128-9233-11e5-8e8a-a2df2a44f698.PNG)

**Comment by wmcook32** [comment]
I have the same thing. I'm gonna look and see if there is another similar error and maybe a fix.

**Comment by nar001** [comment]
Actually, if I ignore it it just launches my instance of MO I use for New Vegas for some reason !

**Comment by TanninOne** [comment]
hrm, delete plugins\pycfg.py and try again.

**Comment by wmcook32** [comment]
that stopped that issue TanninOne. Loaded right up. Should MO login to Nexus/Steam once the info is there or in this Alpha is not working yet?

**Comment by nar001** [comment]
So, weird, but it's still loading the instance of MO I have for NV! (I have three separate folders for Skyrim, Fo4, Fo3 and NV !)

**Comment by 3ventic** [comment]
MO seems to fail to create the links on my system (or load at all). The [vfs log file](https://i.3v.fi/1448324870.txt). Errors I noticed:

```
02:22:07.029 [D] shell32.dll at f3ff0000
02:22:07.029 [E] failed to stub SHFileOperationA: Function starts on a jump
02:22:07.029 [E] failed to stub SHFileOperationW: Function starts on a jump

02:22:07.029 [D] version.dll at ef1d0000
02:22:07.029 [E] failed to stub GetFileVersionInfoW: Function starts on a jump
02:22:07.029 [E] failed to stub GetFileVersionInfoSizeW: Function starts on a jump
```

The Fallout 4 launcher crashes as a result of a stack overflow in usvfs_x64.dll:

> Unhandled exception at 0x00007FFFDF4BA4D4 (usvfs_x64.dll) in Fallout4Launcher.exe: 0xC00000FD: Stack overflow (parameters: 0x0000000000000001, 0x0000000000053FE8).

**Comment by LawLeewer** [comment]
If I launch the game through MO without loading any mod, the game crashes after a few seconds.
If I launch the game through MO loading mods, the launcher appears even if I select the "Fallout 4" executable.  If I press 'play' it closes and the launcher opens  again, not booting the game at all  :/

**Comment by RERLTuna** [comment]
Installing the Fallout Texture Overhaul - Pip-Boy All in One, which has a menu displaying options, got this warning: [W] libpng warning: iCCP: known incorrect sRGB profile

That was for about 25 lines.

Also got a few Geometry warnings like this:
[W] QWindowsWindow::setGeometryDp: Unable to set geometry 200x152+1822+981 on QWidgetWindow/'QProgressDialogClassWindow'. Resulting geometry:  208x152+1822+981 (frame: 12, 52, 12, 12, custom margin: 0, 0,...

But that's been coming up with other mods as well.

Loaded with a few mods, started up FO4 ok, got to the start menu and went to enter on Continue and it crashed, unchecked a few mods, crashes continued, then unchecked all the mods, and the game would just launch into a black screen and discontinue responding. When I switch to load the Launcher first it loads the launcher fine, but I select play and the same thing happens.

**Comment by TheGreenDragon** [comment]
Got it loading the game with 8 mods currently. Ran around for about 30 minutes with no issues. Ini editor is not functional, MO is not logging into Nexus for me yet, and I also get the above mentioned Python error. Also had to white list the MO directory in Avast 2015 or MO would not launch at all.

**Comment by nekoyoubi** [comment]
Using the shiny new alpha, I set my profile up, moved all of my mods over (removed from NMM and re-downloaded/installed), and everything is going pretty well. I have the Python config script error as well, but that's to be expected.

Biggest problem I have is that local saves do not appear to be working for me. As soon as I moved the global toon and files local, I launch the game and there are no saves; move them back to global - saves are back. As of right now (just as a test), I have my toon in global (so I can play), but have the local saves checkbox ticked to see if any new saves would make it to the profile. Nope. Any chance I just missed a memo and profiles do not yet support local saves?

Other than that everything is peachy. I've had the occasional issue while downloading ('blank response' cited), but I figure those are either issues with the Nexus, issues with the NMM-registered file, or just not a big enough deal to worry about since manually downloading the files has the same effect due to downloading into MO2's downloads folder. ;)

Also, I put together a little splash image I thought you might like. If so, feel free to use it; if not, I understand - just an option. http://i.imgur.com/XeYIKi2.png  (Mock: http://i.imgur.com/zNdDsUI.jpg )

**Comment by GrantSP** [comment]
I much prefer that splash to the current one. The current one makes me think of all those bad heavy metal album covers from the '80s! ;)

**Comment by strangerism** [comment]
I am wondering if there is any problem with using Windows 10, I having trouble launching mods with MO2 with crashes at startup. I have logged an issue with my test cases, but I wonder maybe it's something on my end. Do we need to install any visual c++ redist?

**Comment by nekoyoubi** [comment]
@strangerism My testing is being done on Windows 10, so I don't think there's an issue there.

@GrantSP Hey, those were great album covers, thank you! ;p

**Comment by TheGreenDragon** [comment]
@strangerism  I'm running 10 Pro x64 and it is running well for me. Other than the known issues at this time. What mods are you using? Can you launch FO4 through MO with no mods installed?

**Comment by LawLeewer** [comment]
For some reason, everything is running fine today.  It wasn't working yesterday and now I can play the game with mods installed and everything, I don't remember doing anything to fix the problem tho...  Windows 10 pro x64

**Comment by strangerism** [comment]
@TheGreenDragon  I have managed to install 40 small mods, mainly textures animation and sound. No plugin at this time, it's by choice as I don't want to risk my saves on first run. I have troubles installing a couple of mods with lots of texture files (i.e Optimized Vanilla Textures) that will make the game crash at the main menu. Another issue is that disabling the mod won't work, upon launching the game there will be missing texture or crashes, if I want to disable a mod I need to restart MO and then launch the game.

**Comment by wmcook32** [comment]
i have yet to get MO to load the game without crashing instantly. Is anyone by chance running an ENB. That is at the moment the only thing I am running with the game.

**Comment by NoShotz** [comment]
I am running ENBBoost and it is working fine with 26 mods installed

**Comment by wmcook32** [comment]
hrm i'm running windows 7 64bit os. but the game refuses to load through MO.

**Comment by Newyears1978** [comment]
Same issue as wmcook32 - when I launch through MO it gets almost to the menu then crashes to desktop. (with or without mods activated)

**Comment by strangerism** [comment]
using enb and shadowboost, no problems for me

**Comment by nekoyoubi** [comment]
Oddly, now I'm noticing crashes on launch as well, but they are always very quick and are remedied by just running it a second time.

**Comment by MrJelle** [comment]
Windows 7 64-bit, had the Python error on start-up which is gone since removing the relevant file, still getting CTDs and the general feeling like the virtualization isn't happening as it should. Registered to provide another log, let me know if I can do/test anything else.


[usvfs-2015-11-25_14-06-21.txt](https://github.com/TanninOne/modorganizer/files/43917/usvfs-2015-11-25_14-06-21.txt)

**Comment by raydleemsc** [comment]
Count me in for beta testing -
http://www.nexusmods.com/fallout4/users/26883859/? - qa and testing was
part of my masters, anyway

Raymond D. Lee, M.Sc.

**Comment by MitchellMarx** [comment]
I am seeing the same issue as 3ventic with
enbseries_fallout4_v0283
ReShade Framework 1.1.0
VOGUE ENB v0.09 - STANDARD-42-v0-09

**Comment by zSoulweaver** [comment]
Yeah not able to run the game at all through MO even with a clean directory and no mods. I also have an issue where when clicking 'Download with NMM' on the nexus, Mod Organiser will stop working, but apparently the main program is still running and it downloads fine. Mods install fine, just can't actually run the game. The Fallout 4 Launcher just freezes too if I run it from MO.

Windows 10

**Comment by eskachig** [comment]
I seem to have a different problem than most - Fallout runs just fine for me, but none of the mods show up. Not sure what to do, but looking forward to everything getting sorted out. MO is a godsend to modding.

**Comment by nekoyoubi** [comment]
@eskachig If you're using the current Fallout 4 Beta (the opt-in on Steam), then you will not have access to mods.

**Comment by NoShotz** [comment]
@nekoyoubi Its not just that, the plugins.txtand loadorder.txt arent being virtually transfered to the folder in appdata

**Comment by strangerism** [comment]
dumb question to Tannin, is there a possibility that MO would change, write or corrupt the game archives? I know this is not what MO is supposed to do but since testing MO I had many instances of CDT while saving, loading saves and zoning. It seems I have solved my constant zoning CDT with any saves I had by checking game cache in steam which prompt to download 1.4 GB refreshing the following files Fallout4 - Meshes.ba2, Fallout4 - Misc.ba2, Fallout4 - Interface.ba2. On the other end I don't believe this is the case as I did a fresh game reinstall from a steam backup of the game this morning, but honestly I can't remember if I tested MO after that. Could be that the backup was borked also. Surely since I started testing MO my game has been unstable, whether using MO or another Mod Manager or just playing vanilla. This is not blaming MO in the first place, I am just trying to troubleshoot these issues which have cropped up in the last couple of days while I had before that 160 hours with out any problems. Honestly I already know the answer but some kind of confirmation and ruling out MO would help me focus in other areas or things I have done that brought me in this state. Lets hope now that steam fixed it, I need play test a bit.

**Comment by NoShotz** [comment]
@strangerism, i don't think that would be caused by MO, as it doesn't touch that stuff. Also, i am running with around 28 mods (loose and esp) and i haven't had any CTD's with those mods.

**Comment by TanninOne** [comment]
@strangerism No, this isn't possible, MO only affects how/where files are found, it doesn't care for nor affect their content.

**Comment by NoShotz** [comment]
@TanninOne, and idea why the plugins.txt and loadorder.txt arent being "transfered" to the folder in appdata?

**Comment by strangerism** [comment]
well I was quite sure anyway. thanks for the clarification.

@NoShotz can you share your mod list?

**Comment by NoShotz** [comment]
@strangerism currently at school, will share when i get home

**Comment by NoShotz** [comment]
@strangerism These are the mods i have, http://pastebin.com/PEAjWwCU

**Comment by strangerism** [comment]
pretty much the same mods, I don't have any esp though.  

http://pastebin.com/LMv0XT5w

Never had an issue with them since game released until last couple of days when everything went AWOL.

I had a weird occurrence where I was loading the game near USS Constitution and the water level of the river was way way below that normal to the point that boats were all floating mid air. How such thing could happen with just textures and sounds file I have no idea. 

I recovered from it somehow but I don't remember how I did it.  Yet I still had repeatable crashes zoning to Diamond city and nearby. Thankfully since I verified the game cache downloaded those files things are better now but I am playing just vanilla for the time being.

**Comment by RERLTuna** [comment]
Trying it out again, figured out I had plugins.txt as read only for some reason in my appdata, so that may have caused problems. I then figured out every time I enable/disable a mod(s) I have to restart MO for them to take affect or work correctly, then again it seems to be spotty that mods will even load at all if I restart MO. Will keep testing.

**Comment by strangerism** [comment]
I feel weird that MO needs to write plugins.txt in AppData, should't this file be taken care by the virtualization as well?

**Comment by TanninOne** [comment]
@NoShotz it's an alpha
@strangerism it's an alpha

**Comment by strangerism** [comment]
@Tannin I am aware just sprouting some thoughts :)

still I am am not clear if, at the moment, editing plugins.txt in AppData makes a difference or not.

**Comment by TanninOne** [comment]
It should. Mo is currently just not applying the plugins.txt from its profile so the global one is used. I'll probably relase a new alpha tomorrow to fix this and several other issues that came up.

**Comment by WereWookiee** [comment]
Hi!  Love this program.  Waiting for MO compatibility before I dig into modding fallout 4, after using it with FO3 and FONV it's simply indispensable.

I'm on Windows 7 x64 and when I try and launch fallout 4 using your mod organizer alpha it launches, barely gets to the title screen, then I crash to desktop.

Also, on launching Mod Organizer, I get this error window to popup:

"failed to initialize plugin C:/Games/Steam Games/steamapps/common/Fallout 4/Mod Organizer/plugins/pyCfg.py: An unexpected C++ exception was thrown in python code"

**Comment by ThosRTanner** [comment]
That last as has been stated before is a known issue and nothing to worry about

**Comment by RestedWicked** [comment]
like someone said up there, it's crashing with every large texture mod. other than that the virtualization seems to be working fine.
anyone got it to load with large mods? like texture optimization project (~5.5gb uncompressed currently)

**Comment by strangerism** [comment]
I manage to load the Optimized Vanilla Texture bsa just like you, but as you said it's impossible to tell if the texture are loaded because they are the vanilla just optimized, At least it doesn't crash as it did with the loose files.

**Comment by LawLeewer** [comment]
@strangerism I'm pretty sure the OVT bsa is loading.  My gaming was lagging pretty heavily before I loaded it, and the mod solved the problem.

**Comment by TanninOne** [comment]
many of the reported issues should be fixed in alpha 2, please update the issues where things have changed.

**Comment by RERLTuna** [comment]
So far so good, still got the pyCfg.py error but thats fine. Everything else seems to be working out ok. Have not tried changing any of the inis through MO. Will report back once I do. Fantastic job Tannin. Keep up the great work!

**Comment by strangerism** [comment]
overall beta2 it has run smooth for me (Windows 10 x64). Great Success!!

**Comment by pman11** [comment]
Good job Tannin, keep it up.

**Comment by pralima87** [comment]
Testing alpha2 and is working quite well.
just a pyCfg.py error when the app starts but so far is working quite well.

**Comment by strangerism** [comment]
Just a sticky, seems we don't need to change sResourceDataDirsFinal to enable loose files after all

Others have posted confirmation that doing so works as well
sResourceDataDirsFinal=
bInvalidateOlderFiles=1

(ref: http://afkmods.iguanadons.net/index.php?/topic/4316-fallout-4-useful-ini-settings/)

I wasn't unable to try yet as MO2 overrides the profile inis every time I launch the game and I am lazy enough now to go back to manual modding :)

**Comment by darksynth0** [comment]
Please count me in for the Beta 2 testing.
http://www.nexusmods.com/fallout4/users/1191959/?

darksynth0@gmail.com

**Comment by J1mB091** [comment]
The installation scrip [NewDialog](http://www.nexusmods.com/fallout4/mods/619) uses doesn't work with alpha 2.
Seems xml fomod scripts are working but cs ones don't.
Forcing the external fomod installer doesn't work either.

**Comment by HaroldHearthrow** [comment]
Fail Fallout 4 Executable Launch
-------------------------------------------------------------------------------------------------------------------------------
Automatic Plugin Reset
-------------------------------------------------------------------------------------------------------------------------------
Fix - Well At Least for Me.
-------------------------------------------------------------------------------------------------------------------------------
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1)Launch Steam & FO4-ModOrganizer(Admin)
2)Run "Fallout Launcher"
3)Set your preferences or settings. Then Save.
4)Exit Launcher
*All your plugins will automatically be unticked*
5)Right-click on the plugin box/area/space & Click 'Enable All Plugins'
6)Hover mouse to the left where the installed mods are displayed. Right-click the area and select 'Refresh'
7)Select Fallout 4, RUN, & Enjoy! Happy Modding.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
If this helps, please comment back and yeah. Hope it helps.
-------------------------------------------------------------------------------------------------------------------------------
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HaroldHearthrow

**Comment by Sharlikran** [comment]
I am not sure where to ask this @TanninOne but will Mod Organizer be able to use 32 bit apps like FO4Edit or Wrye Flash.  I know Wrye Flash will not be upgraded to a 64 bit app.  FO4Edit probably won't be updated either because some of the libraries used with FO4Edit are 32 bit and the Delphi compiler needs all the libraries to be 64 bit in order to compile a 64 bit version.  So FO4Edit probably won't become a 64 bit app either.

**Comment by TanninOne** [comment]
Alpha 3 is up with many bugfixes (mostly internal to the vfs).

**Comment by nekoyoubi** [comment]
@TanninOne  So I guess that's a no, for using the splash I made you, then? /sadface

**Comment by TanninOne** [comment]
@nekoyoubi The splash you made looks very cool. The reason I'm not using it is that it's so fallout specific. One of the big themes in MO2 is the ability to support non-gamebryo games so I'm hoping for plugins for other games. Therefore I'd rather use a game-neutral splash. However, I'm also considering displaying different splashes depending on the selected game, but this is not currently implemented.

**Comment by nekoyoubi** [comment]
Thanks for the response, and I understand completely. Just know that the Fallout-style was in homage of FO4 support, more so than an attempt to tie MO2 to it. But I do understand entirely, and <3 you all the same. ;p  Though, now that you mention separate splashes for games, is that something that I may be able to help with by chance? Also, I'm a bit confused about your "non-gamebryo"-comment; didn't Bethesda drop Gamebryo when they made Creation?

**Comment by TanninOne** [comment]
Re splashes: sure. I'm no artist so I likely wont create thr splashes myself but include contributions. But please let me first verify its possible.

Re gamebryo: the engine covers many things: graphics, physics, file formats, scripting,... Bethesda changed the graphics and sctipting engine but the file formats (esps, bsas, save games) are practically the same for all these games and thus I can use the same code. Adding support for - say Witcher 3 - would require far more work then adding skyrim support did.
And previously this work would habe to happen in the mo core application which is rather complex.
With mo2 the goal is that everything one needs to support a new game can be done in plugins.

**Comment by DarthFalcon** [comment]
Hi, I'm using the latest build and can't even make it to the main menu, with or without mods. It crashes immediately on the opening video. Are there any special tweaks I need to make? Running on Windows 10.
